### PR TITLE
Cli testing

### PR DESCRIPTION
**Pull request by elsapeng**

### CODE REVIEW

File: bin/swagger-project.js [hunk 1]
```diff
@@ -32,7 +34,7 @@ app
   .command('start [directory]')
   .description('Start the project in this or the specified directory')
   .option('-d, --debug <port>', 'start in remote debug mode')
-  .option('-b, --debug-brk <port>', 'start in remote debug mode, wait for debugger connect')
+  .option('-b, --debug-brk <port>', 'tart in remote debug mode, wait for debugger connect')
```
Comment (mohsen1) [inline bin/swagger-project.js]: Typo

File: lib/commands/project/project.js [hunk 2]
```diff
@@ -363,3 +371,60 @@ function spawn(command, options, cwd, cb) {
...
+      return cb(new Error(util.format('Unknown type: %j. Valid types: %s', options.testModule, TESTMODULES.join(', '))));
+    }
+
+    // check if the assertion-format is one of the three
+    if (options.assertionFormat && !ASSERTIONTYPES[options.assertionFormat]) {
+      return cb(new Error(util.format('Unknown type: %j. Valid types: %s', options.assertionFormat, ASSERTIONTYPES.join(', '))));
+    }
+
+    // process the paths option
+    var regStr = '/' + options.pathName + '/';
```
Comment (mohsen1) [inline lib/commands/project/project.js]: `var reg = new RegExp(options.pathName)` should be enough

File: lib/commands/project/project.js [hunk 3]
```diff
@@ -363,3 +371,60 @@ function spawn(command, options, cwd, cb) {
...
+    // get the result
+    var config = {
+      pathName: desiredPaths,
+      testModule: options.testModule,
+      assertionFormat: options.assertionFormat
+    };
+    var result = template.testGen(project.api.swagger, config);
+
+    //output the result
+    var async = require('async');
```
Comment (mohsen1) [inline lib/commands/project/project.js]: move require calls to top of the file

File: package.json [hunk 4]
```diff
@@ -29,7 +29,9 @@
     "nodemon": "^1.3.7",
     "serve-static": "^1.9.2",
     "swagger-editor": "^2.9.2",
-    "swagger-tools": "^0.8.6"
+    "swagger-tools": "^0.8.6",
+    "async": "^1.2.1",
+    "handlebars": "^3.0.3"
```
Comment (mohsen1) [inline package.json]: Why handlebars is required? Isn't it a dependency of the swagger-test-templates module? Also swagger-test-templates module should be listed here

File: lib/commands/project/project.js [hunk 5]
```diff
@@ -363,3 +371,60 @@ function spawn(command, options, cwd, cb) {
...
+    async.series(result, function(item, cb) {
+      var path = item.name;
+      var filename = path.substirng(1) + '.js';
+      fs.writeFile('/test/' + filename, item.test, function (err) {
```
Comment (mohsen1) [inline lib/commands/project/project.js]: `'/test/'` is an absolute path

File: lib/commands/project/project.js [hunk 6]
```diff
@@ -363,3 +371,60 @@ function spawn(command, options, cwd, cb) {
...
+  //read the yaml file and validate it
+  readProject(directory, options, function(err, project) {
+    if (err) { return cb(err); }
+    swaggerSpec.validateSwagger(project.api.swagger, options, cb);
+
+    // get the array of string paths from json object
+    pathList = Object.keys(project.api.swagger.paths);
+
+    //check if the test frame is one of the two
+    if (options.testModule && !TESTMODULES[options.testModule]) {
```
Comment (mohsen1) [inline lib/commands/project/project.js]: `TESTMODULES[options.testModule]` wont work

File: bin/swagger-project.js (hunk 1 above)
Comment (elsapeng) [inline bin/swagger-project.js]: changed.

File: lib/commands/project/project.js (hunk 2 above)
Comment (elsapeng) [inline lib/commands/project/project.js]: changed.

File: lib/commands/project/project.js (hunk 3 above)
Comment (elsapeng) [inline lib/commands/project/project.js]: changed.

File: package.json (hunk 4 above)
Comment (elsapeng) [inline package.json]: add the swagger-test-template in package.jason

File: lib/commands/project/project.js (hunk 5 above)
Comment (elsapeng) [inline lib/commands/project/project.js]: changed to relative path, but i am not sure if it is ok to use relative path?

File: lib/commands/project/project.js (hunk 6 above)
Comment (elsapeng) [inline lib/commands/project/project.js]: changed to use _.includes

**Pull request closed without merging**